docs/graph-algorithms/kruskals-algorithm.mdx

```diff
@@ -0,0 +1,575 @@
+---
+title: Kruskal's Algorithm
+slug: /graph-algorithms/kruskals-algorithm
+---
+## Introduction
+
+Kruskal's Algorithm is a popular algorithm to find out the minimum spanning trees using the greedy approach. This algorithm was described by Joseph Bernard Kruskal, Jr. in 1956. 
+
+Kruskal's algorithm originally places all the vertices of the graph isolated from one another, to create a forest of single node trees, and then merges these trees step by step, combining at every iteration any two of the trees with some edge of the original graph.
+
+## Explanation
+
+- Given an undirected and connected graph `G=(V, E) where, V = vertices; E = edges`, a spanning tree of graph `G` is a tree that includes every vertex and edge of graph `G`. A single graph `G`, can have many spanning trees.
+
+- Each edge of the tree is associated with a `weight`. The sum of the weights of all the edges results in the `cost`. The tree with the `least cost` is known as the `minimum spanning tree`
+
+- Kruskal's algorithm finds the edge with the `least weight` and adds it to the growing spanning tree. Since it takes the most optimal choice (least weight edge) at each stage, it is said to be a `greedy algorithm`.
+
+- Initially, the weights are arranged in `ascending order` and picked out one by one, through the method of `unification`. If the vertices on either side of the edge belong to different subtrees, the subtrees are merged and the edge is added to the answer.
+
+- After iterating through all the edges in sorted order, all the vertices will belong to the same subtree, which will be the minimum spanning tree.
+
+## Algorithm
+- Input the number of `vertices` and `edges`.
+- Create a graph with the specified number of `vertices` and `edges`.
+- Input the `weights` of each edge.
+- Remove any `loops` or `cycles`.
+- Sort the `weights` of the edges in `ascending order`.
+- Add the edges to the result, one by one, in ascending order if they don't form any cycles or loops and merge the subtrees of the vertices on their ends.
+- Print the result.
+
+## Code
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="c"
+  values={[
+    { label: 'C', value: 'c',},
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+    { label: 'Java', value: 'java', },
+  ]
+}>
```

Comment:
Not a review, but just a question: Do we really need all 4 languages here?

Reply:
@HarshCasper I thought we were supposed to include all the languages that was implemented in NeoAlgo. Please let me know if i have to remove any of the languages.